tests/atomics/test_InsertOp.cpp

```diff
@@ -0,0 +1,52 @@
+#include "gtest/gtest.h"
+
+#include "Kokkos_Core.hpp"
+
+#include "kokkos-utils/atomics/InsertOp.hpp"
+
+using execution_space = Kokkos::DefaultExecutionSpace;
+
+/**
+ * @file
+ *
+ * @addtogroup unittests
+ *
+ * **Atomic insertion**
+ *
+ * This group of tests check the behavior of our atomic insert operators.
+ */
+
+namespace Kokkos::utils::tests::atomics
+{
+
+//! @test Check that @ref Kokkos::utils::atomics::InsertMin works as expected.
+TEST(atomics, InsertMin)
+{
+    using view_t = Kokkos::View<int*, execution_space>;
+
+    constexpr size_t trials = 150;
+
+    const execution_space space {};
+
+    view_t data(Kokkos::view_alloc(space, "data"), 2);
+
+    const utils::atomics::InsertMin insert_min {};
+
+    Kokkos::parallel_for(
+        "atomics::InsertMin",
+        Kokkos::RangePolicy<execution_space>(space, 0, trials),
+        KOKKOS_LAMBDA(const typename execution_space::size_type trial)
+        {
+            insert_min.op(data, 0, trials - trial - 2);
```

Comment:
To avoid having to think about mixing signed and unsigned types, we could also declare `trial` as an `int` and do `trial - 2`. but not sure if there's an issue here.

Reply:
It seems that the compiler is fine, even with `-Wall -Wextra -Werror`.

But I've changed to `int` still.

Reply:
OK!